### PR TITLE
Fixes #629

### LINKED ISSUE #629: @Blocking prevents exception from being logged
With a method such as:

```
@Incoming("kafka-something")
@Blocking
public void onMessage(byte[] message) {
    throw new RuntimeException("Fakeception");
}
```

The exception results in the appropriate failure processing, but the actual exception isn't logged anywhere for subsequent investigation.

See https://github.com/quarkusio/quarkus/issues/10253 for full details

### PR DESCRIPTION
**Pull request by kenfinnigan**
- @Blocking and @Incoming method exceptions are now logged
- JUnit 4/5 co-existing now for unit tests
- Add SmallRye Testing to capture log output from a test

### CODE REVIEW

File: smallrye-reactive-messaging-provider/pom.xml [hunk 1]
```diff
@@ -53,6 +53,18 @@
...
+    <dependency>
+      <groupId>io.smallrye.testing</groupId>
+      <artifactId>smallrye-testing-utilities</artifactId>
+      <scope>test</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>org.jboss.logmanager</groupId>
+      <artifactId>jboss-logmanager</artifactId>
+      <version>2.1.17.Final</version>
```
Comment (cescoffier) [inline smallrye-reactive-messaging-provider/pom.xml:65]: Should we declare this version in the root pom? Maybe even the dependency in the dependency management?

File: smallrye-reactive-messaging-provider/src/test/java/io/smallrye/reactive/messaging/blocking/BlockingExceptionTest.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        addBeanClass(IncomingBlockingExceptionBean.class);
+        initialize();
+
+        IncomingBlockingExceptionBean bean = container.getBeanManager().createInstance()
+                .select(IncomingBlockingExceptionBean.class).get();
+
+        await().until(() -> bean.list().size() == 5);
+        assertThat(bean.list()).contains("a", "b", "d", "e", "f");
+
+        //TODO: Assert the log output contains an exception
```
Comment (cescoffier) [inline smallrye-reactive-messaging-provider/src/test/java/io/smallrye/reactive/messaging/blocking/BlockingExceptionTest.java:32]: Is it done?

File: smallrye-reactive-messaging-provider/pom.xml (hunk 1 above)
Comment (kenfinnigan) [reply smallrye-reactive-messaging-provider/pom.xml]: Done

File: smallrye-reactive-messaging-provider/src/test/java/io/smallrye/reactive/messaging/blocking/BlockingExceptionTest.java (hunk 2 above)
Comment (kenfinnigan) [reply smallrye-reactive-messaging-provider/src/test/java/io/smallrye/reactive/messaging/blocking/BlockingExceptionTest.java]: Oops, removed

**Pull request merged**